### ISSUE TITLE
Redo device management

### ISSUE THREAD

**Issue by azertyfun** [FrOSt-Foundation/FrOSt#52]
Current state is : Support of multiple devices is pretty much non existent.

**Comment by azertyfun** [comment]
OK I tried to better things, but it is a complete mess and compilcated the driver writing very much, so here is a proposal for how things would work : 

Userland has NO access to drivers/devices. Apps would communicate with devices through a device manager. From the user perspective, the code would look like this :

``` dasm
set PUSH, DEVICE_GENERICSCREEN
set PUSH, PRINTF
set PUSH, str_hello
set PUSH 0 ; With support for multiple devices at a time, each device has a number, so if you have three keyboards, you can access them with 0, 1, 2. 0xFFFF would be for "All of them"
int INT_DEVICE
```

Any suggestions ?

**Comment by L3nn0x** [comment]
I don't know about the device_genericscreen

For me it would be stdout/stdin and a write function to a specified fd. So we can also write to a file.
Printf would be calling write with stdout

**Comment by azertyfun** [comment]
Yeah, I know, once we have filesystem and stdin/stdout support you can add that to the screen driver, it won't change how the device manager works.
